awxkit/awxkit/cli/client.py

```diff
@@ -290,6 +290,20 @@ def parse_action(self, page, from_sphinx=False):
 
         # determine the awxkit method to call
         action = self.original_action = parsed.pop('action')
+
+        # if creation is specified with --ignore-existing, try to fetch the object instead
+        if action == 'create' and parsed.get('ignore_existing') is True:
+            _filter = UNIQUENESS_RULES.get(self.resource, ['name'])[0]
+            resp = page.get(**{_filter: parsed[_filter]})
```

Comment:
If this only applies for the `create` action, then it seems like it should go somewhere in create-specific methods instead of the general method.

Down below, it could be put in `if self.original_action == 'create':`

You could even avoid the added get request by inspecting the result, which would honestly be more accurate because the uniqueness rules via the CLI are inexact.

Reply:
That's a good idea.  I'll tinker.

Reply:
The duplicate response we get back doesn't actually seem to contain a reference to the duplicate itself (just this):

`{'__all__': ['JobTemplate with this (name) combination already exists.']}`

...so we probably *do* need to actually attempt to fetch by uniqueness rules.

Reply:
I've pushed a version that _doesn't_ add an extra `GET` to each `POST`, though.